File: 007 - Season 02 Fullstack/001 - Bootcamp Ruby/004 Ruby Quest04/ex00/my_string_index.rb
# QWASAR.IO -- my_string_index
# @param {String} param_1 (haystack)
# @param {Character} param_2 (needle)
# @return {integer}
def my_string_index(param_1, param_2)
  param_1.each_char.with_index do |char, index|
    return index if char == param_2
  end
  -1
end

File: 004 - Season 01 Arc 01/Bootcamp C Arc 01/Quest02/ex08/my_string_index.c
/*Create a my_string_index function which takes 2 parameters (haystack and needle) and locates the first occurrence of the character needle in the string haystack and returns the position.

You can think of this function as: is there an L (character) in my string "helLo"?

The objective is to build a loop that has an if statement which returns the characters position when it matches the right character.
*/

int my_string_index(char* param_1, char param_2) {
    int i = 0;

    // Loop through the string until the null terminator '\0'
    while (param_1[i] != '\0') {
        // Check if the current character matches the needle (param_2)
        if (param_1[i] == param_2) {
            return i;  // Return the index of the first occurrence
        }
        i++;
    }

    // If the character is not found, return -1
    return -1;
}
